apps/model-ad/api-next/src/main/java/org/sagebionetworks/model/ad/api/next/service/DiseaseCorrelationService.java

```diff
@@ -40,26 +47,41 @@ public List<DiseaseCorrelationDto> loadDiseaseCorrelations(
       ItemFilterTypeQueryDto.INCLUDE
     );
 
-    List<DiseaseCorrelationDocument> documents;
+    PageRequest pageable = PageRequest.of(pageNumber, pageSize);
+    Page<DiseaseCorrelationDocument> page;
 
     if (effectiveFilter == ItemFilterTypeQueryDto.INCLUDE) {
       if (items.isEmpty()) {
-        return List.of();
+        page = Page.empty(pageable);
+      } else {
+        List<ObjectId> objectIds = ApiHelper.parseObjectIds(items);
+        page = repository.findByClusterAndIdIn(cluster, objectIds, pageable);
       }
-      List<ObjectId> objectIds = ApiHelper.parseObjectIds(items);
-      documents = repository.findByClusterAndIdIn(cluster, objectIds);
     } else {
       if (items.isEmpty()) {
-        documents = repository.findByCluster(cluster);
+        page = repository.findByCluster(cluster, pageable);
       } else {
         List<ObjectId> objectIds = ApiHelper.parseObjectIds(items);
-        documents = repository.findByClusterAndIdNotIn(cluster, objectIds);
+        page = repository.findByClusterAndIdNotIn(cluster, objectIds, pageable);
       }
     }
 
-    return documents
+    List<DiseaseCorrelationDto> diseaseCorrelations = page
+      .getContent()
       .stream()
       .map(diseaseCorrelationMapper::toDto)
       .collect(Collectors.collectingAndThen(Collectors.toList(), List::copyOf));
+
+    PageMetadataDto pageMetadata = new PageMetadataDto()
+      .number(page.getNumber())
+      .size(page.getSize())
+      .totalElements(page.getTotalElements())
+      .totalPages(page.getTotalPages())
+      .hasNext(page.hasNext())
+      .hasPrevious(page.hasPrevious());
+
+    return new DiseaseCorrelationsPageDto()
+      .diseaseCorrelations(diseaseCorrelations)
+      .page(pageMetadata);
```

Comment:
1) DiseaseCorrelationsPageDto extends PageMetadataDto, so you could directly build it instead of creating an instance of PageMetadataDto. See this example: https://github.com/Sage-Bionetworks/sage-monorepo/blob/d9902a6ac12e20fe14b44dcec138f2391f00d06b/apps/bixarena/api/src/main/java/org/sagebionetworks/bixarena/api/service/ExamplePromptService.java#L50-L58

2) Prefer using the `.builder()` pattern for consistency across the products instead of using the `new` operator. See the above example.